### ISSUE TITLE
Does not support validating Enterprise SSO users

### ISSUE THREAD

**Issue by lukasz-mitka** [mszostok/codeowners-validator#207]
**Description**
Validation fails on checking valid enterprise sso users
Example: `@user-name-can-be-very-long_and-has-underscore`

Validation fails after adding SSO user to CODEOWNERS file.

**Expected result**
User should be validated successfully.

**Actual result**
Fails with 

```
Owner @user_id does not look like a GitHub username or team name
```
**Steps to reproduce**
Add `* @user_id` to CODEOWNERS file

**Troubleshooting**
Here's the regex that doesn't match it.
https://github.com/mszostok/codeowners-validator/blob/84b4fa2742240c05af8c81185e91947d90222106/internal/check/valid_syntax.go#L16

**Comment by lukasz-mitka** [comment]
I just noticed I only fixed the `_` part of this issue in #209.
The unlimited username length will still be an issue :(

**Comment by lukasz-mitka** [comment]
@mszostok Could you please create a release with what we have already available?

Since this is a docker action I can't override the version in a reasonable manner.
And I need #209 available.
